geode-lucene/src/main/java/org/apache/geode/cache/lucene/LuceneIndex.java

```diff
@@ -83,4 +84,14 @@
    */
   LuceneSerializer getLuceneSerializer();
 
+  /**
+   * Returns the status of the Lucene Index
+   * true - if indexing is in progress
+   * false - if indexing is complete
+   *
+   * @Return boolean status of the index.
+   */
+
+  boolean isIndexingInProgress();
```

Comment:
I wanted this API to be more precise from a developer's point of view. As in what is the status of the index, isIndexingInProgress will tell reindexing is in progress but isIndexReady()'s boolean answer is like index cannot be ready (maybe because it is defined but not initialized or because it is in process of reindexing). I was thinking maybe we need to add more operation checks in the future and thought that isIndexingInProgress pinpoints directly to the indexing process.